### ISSUE TITLE
How to pass data to guest executable at runtime?

### ISSUE THREAD

**Issue by nineralpha** [Azure/service-fabric-issues#856]
I have a stateless service that spin up a new guest executable on demand.I need to pass some data at service creation time to guest executable.
How can i do that??

**Comment by xunlu** [comment]
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-deploy-existing-app#overview-of-application-and-service-manifest-files

Have you read this?

**Comment by masnider** [comment]
[Environment Variables ](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-how-to-specify-environment-variables) also work

**Comment by nineralpha** [comment]
@xunlu yes already tried Entrypoint and setupEntrypoint. but problem is that they are used to pass hardcoded data. i am using
StatelessServiceDescription serviceDescription = new StatelessServiceDescription()
            {
                ApplicationName = new Uri(m_appName),
                InstanceCount = 1,
                PartitionSchemeDescription = new SingletonPartitionSchemeDescription(),
                ServiceTypeName = serviceType,
                ServiceName = new Uri(serviceName),
                ServicePackageActivationMode = ServicePackageActivationMode.ExclusiveProcess
            };
and then 
await m_fabricClient.ServiceManager.CreateServiceAsync(serviceDescription); 
to create a new guest executable service at run time from a stateful reliable service.
now i have 100's of guest executable services and i have to pass different data to each service
1)i cannot use Entrypoint and setupEntrypoint
2)cannot set environment variable of one service from another.

**Comment by RajeetN** [comment]
How big is the data you want to pass? You can create a new application+Service pair and pass environment variables via ApplicationParameters to override the values when new app is created.

**Comment by nineralpha** [comment]
Each guest executable works on a unique file it downloads from cloud and we have to provide path of that file and also directory structure where it will be saved on local disk.
application+Service is not an option here.

**Comment by usamaazhar** [comment]
@RajeetN Does creating a new application for each service follow the micro-services guidelines for service fabric ? I mean the Gist is to have multiple micro-services in a single application

**Comment by RajeetN** [comment]
SF doesn't have control over how guest executable consumes data so doesn't have mechanism for this. You can create a stateless service and pass the data as initialization parameter to it.
